src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -99,7 +99,7 @@ public abstract partial class TypeDesc
         public override int GetHashCode()
         {
             // Inherited types are expected to override
-            return RuntimeHelpers.GetHashCode(this);
+            throw new NotImplementedException();
```

Comment:
Are we talking about the same thing?

This is what I meant:

```csharp
abstract class Foo
{
    public abstract override int GetHashCode();
}

class Bar : Foo
{
    public override int GetHashCode()
    {
        return 123456;
    }
}

class Program
{
    static void Main(string[] args)
    {
        object o = new Bar();
        Console.WriteLine(o.GetHashCode());
    }
}
```
It does call Object::GetHashCode and it will print 123456.